### PR TITLE
Indent lambda parameters if parameters wrap

### LINKED ISSUE #8179: Formatter undocumented deviation:
[Black formatting](https://black.vercel.app/?version=stable&state=_Td6WFoAAATm1rRGAgAhARYAAAB0L-Wj4AEtAIddAD2IimZxl1N_WlbvK5V8G17R4xukEnqOHRjoYorGA3_x5d79TklUYR_XtF6MM4d5srbPV3-N4pXQ1SwF_YZZxkksYNb5k5P7chlg8SSpdJuccpW53I3EE2uXGr5zBl09V3WMIRMwonXPxdgmzF3oapEcKjKrbacGLJ8ijg5fK84E8SAp_bjGAAAA8DH6Gx1SCp8AAaMBrgIAAHUkjc-xxGf7AgAAAAAEWVo=):
```
def a():
    return b(
        c,
        d,
        e,
        f=lambda self, *args, **kwargs: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa(
            *args, **kwargs
        ),
    )
```

[Ruff formatting](https://play.ruff.rs/?secondary=Format):
```
def a():
    return b(
        c,
        d,
        e,
        f=lambda self,
        *args,
        **kwargs: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa(*args, **kwargs),
    )
```

In addition to being incompatible, this formatting seems very strange and not very readable, as the `*args` and `**kwargs` suddenly look like they are arguments to `b`.

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

A non-black compatible approach to https://github.com/astral-sh/ruff/issues/8179

The black-compatible formatting would enforce spaces when formatting parameters with the `ParameterParentheses::Never`. However, this doesn't work when using comments (which black doesn't support [playground](https://black.vercel.app/?version=stable&state=_Td6WFoAAATm1rRGAgAhARYAAAB0L-Wj4ACOAGBdAD2IimZxl1N_WhQ8R5sgERcjE4CA79XBQnYhkNIGX378uPpzJVCf3opcwNZ5N4_s5GR-d6S0M12O66BmJ9d2BIpKmRsn-5XXsdd5l9RAxvUQiFV4AOsopTANF-mxwYxCAADlTL2qgfqd_wABfI8BAAAAQVm_NLHEZ_sCAAAAAARZWg==)?)

This PR implements two changes (we may want to split it in two):

a) It indents the lambda parameters if they don't fit on a single line: 

```python
def a():
    return b(
        c,
        d,
        e,
        f=lambda 
            self,
            *args,
            **kwargs
        : aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa(*args, **kwargs),
        b=d,
        e=f,
    )

```

b) Preview style that parenthesizes the lambda body if it expands. 

```diff
         return {
             **super().get_event_triggers(),
             EventTriggers.ON_CHANGE: (
-                lambda e0: [
-                    Var.create_safe(f"{e0}.map(e => e.value)", _var_is_local=True)
-                ]
-                if self.is_multi.equals(Var.create_safe(True))
-                else lambda e0: [e0]
+                lambda e0: (
+                    [Var.create_safe(f"{e0}.map(e => e.value)", _var_is_local=True)]
+                    if self.is_multi.equals(Var.create_safe(True))
+                    else lambda e0: [e0]
+                )
             ),
         }
```

## Alternatives

Favor black compatibility and enforce using `space` in the parameters formatting when the `ParameterParentheses::Never` is used. Requires finding a comment placement that produces stable results. 

Unclear how we want to support 

```python
(
    lambda
    x,
    # comment
    y:
    z
)
```

## Notes

It's probably worth to split out the body changes. They seem to generally improve readability

## Test Plan

The non-preview changes don't seem to change the similarity index (at least after only indenting if there are 2 or more parameters). 

The preview changes regress the similarity index across the board

Main
| project        | similarity index  | total files       | changed files     |
|----------------|------------------:|------------------:|------------------:|
| cpython        |           0.75804 |              1799 |              1648 |
| django         |           0.99984 |              2772 |                34 |
| home-assistant |           0.99963 |             10596 |               143 |
| poetry         |           0.99925 |               317 |                12 |
| transformers   |           0.99967 |              2657 |               322 |
| twine          |           1.00000 |                33 |                 0 |
| typeshed       |           0.99980 |              3669 |                18 |
| warehouse      |           0.99977 |               654 |                13 |
| zulip          |           0.99970 |              1459 |                21 |

Preview

| project        | similarity index  | total files       | changed files     |
|----------------|------------------:|------------------:|------------------:|
| cpython        |           0.75804 |              1799 |              1648 |
| django         |           0.99978 |              2772 |                40 |
| home-assistant |           0.99957 |             10596 |               174 |
| poetry         |           0.99925 |               317 |                12 |
| transformers   |           0.99965 |              2657 |               328 |
| twine          |           1.00000 |                33 |                 0 |
| typeshed       |           0.99980 |              3669 |                18 |
| warehouse      |           0.99968 |               654 |                16 |
| zulip          |           0.99970 |              1459 |                21 |

Here a few examples

```diff
index 9aa78a281f..499bc0dfb9 100644
--- a/django/db/models/expressions.py
+++ b/django/db/models/expressions.py
@@ -366,21 +366,21 @@ class BaseExpression:
         internal_type = field.get_internal_type()
         if internal_type == "FloatField":
             return (
-                lambda value, expression, connection: None
-                if value is None
-                else float(value)
+                lambda value, expression, connection: (
+                    None if value is None else float(value)
+                )
             )
         elif internal_type.endswith("IntegerField"):
             return (
-                lambda value, expression, connection: None
-                if value is None
-                else int(value)
+                lambda value, expression, connection: (
+                    None if value is None else int(value)
+                )
             )
         elif internal_type == "DecimalField":
             return (
-                lambda value, expression, connection: None
-                if value is None
-                else Decimal(value)
+                lambda value, expression, connection: (
+                    None if value is None else Decimal(value)
+                )
             )
         return self._convert_value_noop
 
index 889c1cfe84..8555f08cb3 100644
--- a/django/contrib/gis/db/models/fields.py
+++ b/django/contrib/gis/db/models/fields.py
@@ -48,9 +48,9 @@ def get_srid_info(srid, connection):
     alias, get_srs = (
         (
             connection.alias,
-            lambda srid: SpatialRefSys.objects.using(connection.alias)
-            .get(srid=srid)
-            .srs,
+            lambda srid: (
+                SpatialRefSys.objects.using(connection.alias).get(srid=srid).srs
+            ),
         )
         if SpatialRefSys
         else (None, SpatialReference)


index dc857055b1..b45a8bb10e 100644
--- a/django/contrib/admin/tests.py
+++ b/django/contrib/admin/tests.py
@@ -110,8 +110,9 @@ class AdminSeleniumTestCase(SeleniumTestCase, StaticLiveServerTestCase):
         Block until the  page is ready.
         """
         self.wait_until(
-            lambda driver: driver.execute_script("return document.readyState;")
-            == "complete",
+            lambda driver: (
+                driver.execute_script("return document.readyState;") == "complete"
+            ),
             timeout,
         )
 
@@ -196,8 +197,8 @@ class AdminSeleniumTestCase(SeleniumTestCase, StaticLiveServerTestCase):
             # to be the case.
             with self.disable_implicit_wait():
                 self.wait_until(
-                    lambda driver: not driver.find_elements(
-                        By.CSS_SELECTOR, options_selector
+                    lambda driver: (
+                        not driver.find_elements(By.CSS_SELECTOR, options_selector)
                     )
                 )

index 9855c318be..fd7770eb2e 100644
--- a/django/test/testcases.py
+++ b/django/test/testcases.py
@@ -1470,8 +1470,10 @@ def skipIfDBFeature(*features):
 def skipUnlessDBFeature(*features):
     """Skip a test unless a database has all the named features."""
     return _deferredSkip(
-        lambda: not all(
-            getattr(connection.features, feature, False) for feature in features
+        lambda: (
+            not all(
+                getattr(connection.features, feature, False) for feature in features
+            )
         ),
         "Database doesn't support feature(s): %s" % ", ".join(features),
         "skipUnlessDBFeature",
@@ -1481,8 +1483,10 @@ def skipUnlessDBFeature(*features):
 def skipUnlessAnyDBFeature(*features):
     """Skip a test unless a database has any of the named features."""
     return _deferredSkip(
-        lambda: not any(
-            getattr(connection.features, feature, False) for feature in features
+        lambda: (
+            not any(
+                getattr(connection.features, feature, False) for feature in features
+            )
         ),
         "Database doesn't support any of the feature(s): %s" % ", ".join(features),
         "skipUnlessAnyDBFeature",

index b9a7e4cd64..a69ebd31cb 100644
--- a/tests/absolute_url_overrides/tests.py
+++ b/tests/absolute_url_overrides/tests.py
@@ -29,8 +29,9 @@ class AbsoluteUrlOverrideTests(SimpleTestCase):
 
         with self.settings(
             ABSOLUTE_URL_OVERRIDES={
-                "absolute_url_overrides.testb": lambda o: "/overridden-test-b/%s/"
-                % o.pk,
+                "absolute_url_overrides.testb": lambda o: (
+                    "/overridden-test-b/%s/" % o.pk
+                ),
             },
         ):

index 4d85d15065..9aec0a68e8 100644
--- a/tests/contenttypes_tests/test_views.py
+++ b/tests/contenttypes_tests/test_views.py
@@ -151,9 +151,9 @@ class ContentTypesViewsSiteRelTests(TestCase):
         The shortcut view works if a model's ForeignKey to site is None.
         """
         get_model.side_effect = (
-            lambda *args, **kwargs: MockSite
-            if args[0] == "sites.Site"
-            else ModelWithNullFKToSite
+            lambda *args, **kwargs: (
+                MockSite if args[0] == "sites.Site" else ModelWithNullFKToSite
+            )
         )
 
         obj = ModelWithNullFKToSite.objects.create(title="title")
@@ -173,9 +173,9 @@ class ContentTypesViewsSiteRelTests(TestCase):
         found in the m2m relationship.
         """
         get_model.side_effect = (
-            lambda *args, **kwargs: MockSite
-            if args[0] == "sites.Site"
-            else ModelWithM2MToSite
+            lambda *args, **kwargs: (
+                MockSite if args[0] == "sites.Site" else ModelWithM2MToSite
+            )
         )
```

Most of these seem clear improvements to me.

### CODE REVIEW

**Comment by MichaReiser** [comment]
Current dependencies on/for this PR:
* main
  * **PR #8465** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/8465" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/astral-sh/ruff/8465?utm_source=stack-comment).

File: crates/ruff_python_formatter/tests/snapshots/format@expression__lambda.py.snap [hunk 1]
```diff
@@ -289,7 +273,9 @@ a = (
 
 # lambda arguments don't have parentheses, so we never add a magic trailing comma ...
 def f(
-    aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa: bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb = lambda x: y,
+    aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa: bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb = lambda 
+        x
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/tests/snapshots/format@expression__lambda.py.snap:277]: Not sure if this looks better

File: crates/ruff_python_formatter/tests/snapshots/format@expression__lambda.py.snap [hunk 2]
```diff
@@ -336,36 +322,42 @@ lambda a, /, c: a
...
-    # comment
-    *x: x
+    lambda 
+        # comment
+        *x
+    : x
 )
 
-lambda: (  # comment
+lambda: (
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/tests/snapshots/format@expression__lambda.py.snap:359]: This was better before

File: crates/ruff_python_formatter/tests/snapshots/format@expression__lambda.py.snap [hunk 3]
```diff
@@ -289,7 +368,9 @@ a = (
 
 # lambda arguments don't have parentheses, so we never add a magic trailing comma ...
 def f(
-    aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa: bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb = lambda x: y,
+    aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa: bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb = lambda 
+        x
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/tests/snapshots/format@expression__lambda.py.snap:395]: Undecided if this looked better before.

File: crates/ruff_python_formatter/tests/snapshots/format@expression__lambda.py.snap [hunk 4]
```diff
@@ -399,20 +485,377 @@ lambda: (  # comment
...
+
+lambda self, araa, kkkwargs=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa(
+    *args, **kwargs
+), e=1, f=2, g=2: d
+
+
+# Leading
+lambda x: (
+    lambda y: (
+        lambda z: (
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/tests/snapshots/format@expression__lambda.py.snap:844]: This seems better to me

File: crates/ruff_python_formatter/tests/snapshots/format@expression__lambda.py.snap [hunk 5]
```diff
@@ -399,20 +485,377 @@ lambda: (  # comment
...
+            + y
+            + y
+            + z
+        )
+    )  # Trailing
+)  # Trailing
+
+
+# Leading
+lambda x: (
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/tests/snapshots/format@expression__lambda.py.snap:874]: This adds parentheses, but I don't mind them

File: crates/ruff_python_formatter/tests/snapshots/format@expression__lambda.py.snap [hunk 6]
```diff
@@ -424,21 +508,408 @@ lambda: (  # comment
...
+        : aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa(*args, **kwargs),
+    )
+
+
+def a():
+    return b(
+        c,
+        d,
+        e,
+        f=lambda
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/tests/snapshots/format@expression__lambda.py.snap:619]: It looks okay.... Parentheses would make it look so much better

File: crates/ruff_python_formatter/src/comments/placement.rs [hunk 7]
```diff
@@ -1805,18 +1805,6 @@ fn handle_lambda_comment<'a>(
     locator: &Locator,
 ) -> CommentPlacement<'a> {
     if let Some(parameters) = lambda.parameters.as_deref() {
-        // Comments between the `lambda` and the parameters are dangling on the lambda:
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/src/comments/placement.rs:1808]: We should revert this change but it triggers an instability that I haven't yet been able to figure out

**Comment by zanieb** [comment]
@MichaReiser are we trying to decide if this is _better_ than before or if we have correctly conformed to Black's preview style?

**Comment by MichaReiser** [comment]
Parenthesizing the body is probably less controversial. It might be worth avoiding to parenthesize the body if it has parentheses itself. But parenthesizing the body otherwise fits with the formatting that copilot would choose.
